ocaml/idl/datamodel.ml

```diff
@@ -8638,6 +8649,103 @@ let vgpu_type =
 		]
 	()
 
+(** PVS
+ * http://www.uk.xensource.com/~robho/pvs-direct/#external-interfaces-xenapi-pvs_farm-new-class
+ *)
+
+module PVS_farm = struct
+	let lifecycle = [Prototyped, rel_dundee_plus, ""]
+	let refset x	= Set(Ref x)
+
+	let introduce = call
+		~name:"introduce"
+		~doc:"introduce new PVS farm"
+		~result:(Ref _pvs_farm, "the new PVS farm")
+		~params:
+		[ String,"name","name of the PVS farm"
+		; Ref _sr,"cache_sr","SR used for caching"
+		]
+		~lifecycle:lifecycle
+		~allowed_roles:_R_POOL_OP
+		()
+
+	let set_name = call
+		~name:"set_name"
+		~doc:"Update the name of the PVS farm"
+		~params:
+		[ Ref _self, "self", "this PVS farm"
+		; String, "value", "name to be used"
+		]
+		~lifecycle:lifecycle
+		~allowed_roles:_R_POOL_OP
+		()
+
+	let set_cache_sr = call
+		~name:"set_cache_sr"
+		~doc:"Update the cache SR for the proxies on the farm"
+		~params:
+		[ Ref _self, "self", "this PVS farm"
+		; Ref _sr, "value", "SR to be used"
+		]
+		~lifecycle:lifecycle
+		~allowed_roles:_R_POOL_OP
+		()
+
+	let forget = call
+		~name:"forget"
+		~doc:"forget a PVS farm"
+		~params:
+		[ Ref _self, "self", "this PVS farm"
+		]
+		~lifecycle:lifecycle
+		~allowed_roles:_R_POOL_OP
+		()
+
+end
+
+let pvs_farm =
+		let null_ref = Some (VRef (Ref.string_of Ref.null)) in
+		let null_str = Some (VString "") in
+		create_obj
+		~name: _pvs_farm
+		~descr:"machines serving blocks of data for provisioning VMs"
+		~doccomments:[]
+		~gen_constructor_destructor:false
+		~gen_events:true
+		~in_db:true
+		~lifecycle:PVS_farm.lifecycle
+		~persist:PersistEverything
+		~in_oss_since:None
+		~messages_default_allowed_roles:_R_POOL_OP
+		~contents:
+		[ uid     _pvs_farm ~lifecycle:PVS_farm.lifecycle
+
+		; field   ~qualifier:StaticRO ~lifecycle:PVS_farm.lifecycle
+		          ~ty:String "name" ~default_value:null_str
+		          "Name of the PVS farm. Must match name configured in PVS"
+
+		; field   ~qualifier:StaticRO ~lifecycle:PVS_farm.lifecycle
+		          ~ty:(Ref _sr) "cache_sr" ~default_value:null_ref
+		          "The SR used by PVS proxy for the cache"
+
+		(*
+		; field   ~qualifier:StaticRO ~lifecycle:PVS_farm.lifecycle
```

Comment:
OK, just wanted to check. I only took a glance at the combined diff so missed
your commit message. If there's a good reason to have them in and you'll be
un-commenting them soon then that's cool.


On Fri, 10 Jun 2016 at 02:25, Christian Lindig wrote:
> > +		~persist:PersistEverything
> > +		~in_oss_since:None
> > +		~messages_default_allowed_roles:_R_POOL_OP
> > +		~contents:
> > +		[ uid     _pvs_farm ~lifecycle:PVS_farm.lifecycle
> > +
> > +		; field   ~qualifier:StaticRO ~lifecycle:PVS_farm.lifecycle
> > +		          ~ty:String "name" ~default_value:null_str
> > +		          "Name of the PVS farm. Must match name configured in PVS"
> > +
> > +		; field   ~qualifier:StaticRO ~lifecycle:PVS_farm.lifecycle
> > +		          ~ty:(Ref _sr) "cache_sr" ~default_value:null_ref
> > +		          "The SR used by PVS proxy for the cache"
> > +
> > +		(*
> > +		; field   ~qualifier:StaticRO ~lifecycle:PVS_farm.lifecycle
> 
> This is what the comment above refers to: _Two fields are not yet defined because they depend on other, not-yet defined classes._. I can take them out but they are part of the design.
> 
> ---
> You are receiving this because you commented.
> Reply to this email directly or view it on GitHub:
> https://github.com/xenserver/xen-api/pull/95/files/52ff298c9e1e3de7ec83dfdddb479e2d5fc9068e#r66585104
Si.